invenio/templates/configurations/logstash.yaml

```diff
@@ -53,66 +53,125 @@ data:
-            target => "[@metadata]"
+        if [data][upstream_response_time] == "-"{
+          mutate { replace => { upstream_response_time => 0 }}
         }
-        ##############################################################
-        # USER AGENT
-        ##############################################################
-        # add ua/device only if set
-        if [@metadata][ua_device] {
-              mutate {
-                  add_field => { "[data][ua/device]" => "%{[@metadata][ua_device]}" }
-              }
-          }
-        # add ua/family only if set
-        if [@metadata][ua_name] {
-            mutate {
-                add_field => { "[data][ua/family]" => "%{[@metadata][ua_name]}" }
-            }
+        # Enrich with geolocation data.
+        geoip {
+            source => "http_x_forwarded_for" #field name that contains the ip address
+            fields => [ "city_name", "country_name", "country_code2", "continent_code"]
+            target => "data"
+            # Download static database and point to it
         }
-        # add ua/patch only if set
-        if [@metadata][ua_patch] {
-            mutate {
-                add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}.%{[@metadata][ua_patch]}" }
-            }
-        } else {
+
+      useragent {
+          source => "[data][user_agent]"
+          target => "[@metadata]"
+      }
+      ##############################################################
+      # USER AGENT
+      ##############################################################
+      # add ua/device only if set
+      if [@metadata][ua_device] {
             mutate {
-                add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}" }
+                add_field => { "[data][ua/device]" => "%{[@metadata][ua_device]}" }
             }
         }
-        # add ua/os only if set
-        if [@metadata][ua_os] {
-            mutate {
-                add_field => { "[data][ua/os]" => "%{[@metadata][ua_os]}" }
-            }
+      # add ua/family only if set
+      if [@metadata][ua_name] {
+          mutate {
+              add_field => { "[data][ua/family]" => "%{[@metadata][ua_name]}" }
+          }
+      }
+      # add ua/patch only if set
+      if [@metadata][ua_patch] {
+          mutate {
+              add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}.%{[@metadata][ua_patch]}" }
+          }
+      } else {
+          mutate {
+              add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}" }
+          }
+      }
+      # add ua/os only if set
+      if [@metadata][ua_os] {
+          mutate {
+              add_field => { "[data][ua/os]" => "%{[@metadata][ua_os]}" }
+          }
+      }
+      # add a field with the category, based on the url
```

Comment:
From here to 126 is to do the same thing that was done in puppet:
https://gitlab.cern.ch/ai/it-puppet-hostgroup-zenodo/-/blame/qa/code/templates/flume/morphlines/logs.morphline.conf.erb#L72

Note that this implementation has an issue if the grok in line 44 fails. I was thinking about adding a check for the `_grokparsefailure` there. It would be cleaner